lib/themes/dosomething/paraneue_dosomething/js/components/Slideshow.js

```diff
@@ -20,9 +21,16 @@ class Slideshow extends React.Component {
   }
 
   render() {
+    const user = this.props.user;
+    const campaign = this.props.campaign;
+
+    const slides = this.props.slides.map((slide, index) => {
+      return React.createElement(slide, {key: index, campaign: campaign, user: user});
+    });
+
     return (
       <div className="slideshow">
-        {this.props.slides[this.state.activeSlide]}
+        {slides}
```

Comment:
I think it makes sense to only render the visible slide into the Virtual DOM, so we can use [ReactCSSTransitionGroup](https://facebook.github.io/react/docs/animation.html) to handle transitions and our interface stays declarative.